X509: always base64-encode extensions for which _getMapping returns a bool

terrafrost opened the pull request:
Same thing as #628 but against 1.0 instead.

bantu commented on tests/Unit/File/X509/X509Test.php:
```diff
@@ -0,0 +1,60 @@
+
+        $spkac = $x509->loadX509($test);
+
+        $this->assertEquals($a['tbsCertificate']['extensions'][8]['extnValue'], 'MDUwDgYIKoZIhvcNAwICAgCAMA4GCCqGSIb3DQMEAgIAgDAHBgUrDgMCBzAKBggqhkiG9w0DBw==');
```
Swap arguments. $expected comes first.

terrafrost commented:
Replaces #628

bantu commented:
> Btw, I am still curious if it is possible to change the branch a PR is made against. Any ideas?

Unfortunately it's not.

bantu commented on tests/Unit/File/X509/X509Test.php:
```diff
@@ -0,0 +1,60 @@
+
+        $cert = $x509->loadX509($test);
+
+        $this->assertEquals($cert['tbsCertificate']['extensions'][8]['extnValue'], 'MDUwDgYIKoZIhvcNAwICAgCAMA4GCCqGSIb3DQMEAgIAgDAHBgUrDgMCBzAKBggqhkiG9w0DBw==');
```
Please swap the parameters. $expected comes first.

terrafrost commented:
The latest commit addresses your comment.

I'll do the unit test for #625 after this has been merged (rather than creating a second copy of X509Test.php that's pretty much guaranteed to have a merge conflict)

bantu commented:
This looks good. The test fails without the patch. :-)

bantu commented:
Changed class name on merge.
